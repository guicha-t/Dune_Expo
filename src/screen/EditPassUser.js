import React, { Component } from 'react';
import { Alert, Button, TextInput, View, Text, StyleSheet, ScrollView, Image, TouchableOpacity, AsyncStorage } from 'react-native';
import { observer } from 'mobx-react';

import Header from './../global/header/Header';
import Store from './../global/store/Store'

@observer
export default class EditPassUser extends Component {
  constructor(props){
    super(props);
    this.state = {
      oldpassword: '',
      newpassword: '',
      newpasswordbis: '',
    }
  }

  _cancelEdit = async () => {
    this.props.navigation.navigate('Profil');
  };

  _confirmEdit = async () => {
    if (this.state.newpassword === this.state.newpasswordbis)
    {
      fetch('http://176.31.252.134:7001/api/v1/users/changePassword', {
        method: 'PUT',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
          token: Store.Token,
        },
        body: JSON.stringify({
          oldPassword: this.state.oldpassword,
          newPassword: this.state.newpassword,
        }),
      }).then((response) => response.json())
          .then((responseJson) => {
            if (responseJson.status === 200) {
              this.props.navigation.navigate('Profil');
            } else {
              Alert.alert(responseJson.error)
            }
      })
      .catch((error) => {
        console.error(error);
      });

    } else {
      Alert.alert("Pas de concordance")
    }
  };


  render() {
    return(

      <View style={styles.container}>
        <Header navigation={this.props.navigation}/>

        <View style={styles.containerBody}>

          <View style={{height: 30, flexDirection: 'row', alignItems: 'center', justifyContent: 'center', marginBottom: 20}}>
            <View style={{flex: 0.3, alignItems: 'center', paddingBottom: 8}}>
              <Text style={styles.titleInfo}>Ancien</Text>
            </View>
            <View style={{flex: 0.7}}>
              <TextInput
                 style={styles.input}
                 placeholder='Mot de passe'
                 onChangeText={(oldpassword) => this.setState({oldpassword})}
                 secureTextEntry={true}
                 value={this.state.oldpassword}
               />
             </View>
          </View>

          <View style={{height: 30, flexDirection: 'row', alignItems: 'center', justifyContent: 'center', marginBottom: 20}}>
            <View style={{flex: 0.3, alignItems: 'center', paddingBottom: 8}}>
              <Text style={styles.titleInfo}>Nouveau*</Text>
            </View>
            <View style={{flex: 0.7}}>
              <TextInput
                 style={styles.input}
                 placeholder='Mot de passe'
                 onChangeText={(newpassword) => this.setState({newpassword})}
                 secureTextEntry={true}
                 value={this.state.newpassword}
               />
             </View>
          </View>

          <View style={{height: 30, flexDirection: 'row', alignItems: 'center', justifyContent: 'center', marginBottom: 20}}>
            <View style={{flex: 0.3, alignItems: 'center', paddingBottom: 8}}>
              <Text style={styles.titleInfo}>Nouveau*</Text>
            </View>
            <View style={{flex: 0.7}}>
              <TextInput
                 style={styles.input}
                 placeholder='Mot de passe'
                 onChangeText={(newpasswordbis) => this.setState({newpasswordbis})}
                 secureTextEntry={true}
                 value={this.state.newpasswordbis}
               />
             </View>
          </View>
          <Text style={{fontSize: 10}}>* le nouveau mot de passe doit contenir au minimum 8 caract√®res.</Text>

         </View>
         <View style={styles.containerFooter}>
           <Button
             title={'Retour'}
             style={styles.ButtonCo}
             color='#363453'
             onPress={this._cancelEdit}
           />
           <Button
             title={'Valider'}
             style={styles.ButtonCo}
             color='#363453'
             onPress={this._confirmEdit}
           />
         </View>

      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FFF',
    flex: 1
  },
  containerBody: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  input: {
    width: 220,
    height: 44,
    padding: 10,
    borderWidth: 1,
    borderColor: 'black',
    marginBottom: 10,
  },
  titleInfo: {
    color: '#363453',
    fontWeight: 'bold',
    fontSize: 14,
  },
  containerFooter: {
    flexDirection: 'row',
    paddingBottom: 10,
    justifyContent:'space-around',
  },
});
