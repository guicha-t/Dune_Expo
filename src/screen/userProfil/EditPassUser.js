import React, { Component } from 'react';
import { Alert, Button, TextInput, View, Text, StyleSheet, ScrollView, Image, TouchableOpacity, AsyncStorage } from 'react-native';
import { observer } from 'mobx-react';

import Header from './../../global/header/Header';
import Store from './../../global/store/Store'

@observer
export default class EditPassUser extends Component {
  constructor(props){
    super(props);
    this.state = {
      oldpassword: '',
      newpassword: '',
      newpasswordbis: '',
      pwdhide: true,
      opacity: 1,
    }
  }

  _cancelEdit = async () => {
    this.props.navigation.navigate('Profil');
  };

  _confirmEdit = async () => {
    if (this.state.newpassword === this.state.newpasswordbis)
    {
      fetch('http://api.dune-table.com/v1/users/changePassword', {
        method: 'PUT',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
          token: Store.Token,
        },
        body: JSON.stringify({
          oldPassword: this.state.oldpassword,
          newPassword: this.state.newpassword,
        }),
      }).then((response) => response.json())
          .then((responseJson) => {
            if (responseJson.status === 200) {
              this.props.navigation.navigate('Profil');
            } else {
              Alert.alert(responseJson.error)
            }
      })
      .catch((error) => {
        console.error(error);
      });

    } else {
      Alert.alert("Pas de concordance")
    }
  };

  _pwdhide = async () => {
    if (this.state.pwdhide === true) {
      this.setState({'pwdhide':false})
      this.setState({'opacity':0.1})
    } else {
      this.setState({'pwdhide':true})
      this.setState({'opacity':1})
    }
  };



  render() {
    return(

      <View style={styles.container}>
        <Header navigation={this.props.navigation}/>

        <View style={styles.containerBody}>

          <View style={{height: 30, flexDirection: 'row', alignItems: 'center', justifyContent: 'center', marginBottom: 20}}>
            <View style={{flex: 0.3, alignItems: 'center', paddingBottom: 8}}>
              <Text style={styles.titleInfo}>Ancien</Text>
            </View>
            <View style={{flex: 0.7}}>
              <TextInput
                 style={styles.input}
                 placeholder='Mot de passe'
                 onChangeText={(oldpassword) => this.setState({oldpassword})}
                 secureTextEntry={this.state.pwdhide}
                 value={this.state.oldpassword}
               />
             </View>
          </View>

          <View style={{height: 30, flexDirection: 'row', alignItems: 'center', justifyContent: 'center', marginBottom: 20}}>
            <View style={{flex: 0.3, alignItems: 'center', paddingBottom: 8}}>
              <Text style={styles.titleInfo}>Nouveau*</Text>
            </View>
            <View style={{flex: 0.7}}>
              <TextInput
                 style={styles.input}
                 placeholder='Mot de passe'
                 onChangeText={(newpassword) => this.setState({newpassword})}
                 secureTextEntry={this.state.pwdhide}
                 value={this.state.newpassword}
               />
             </View>
          </View>

          <View style={{height: 30, flexDirection: 'row', alignItems: 'center', justifyContent: 'center', marginBottom: 20}}>
            <View style={{flex: 0.3, alignItems: 'center', paddingBottom: 8}}>
              <Text style={styles.titleInfo}>Nouveau*</Text>
            </View>
            <View style={{flex: 0.7}}>
              <TextInput
                 style={styles.input}
                 placeholder='Mot de passe'
                 onChangeText={(newpasswordbis) => this.setState({newpasswordbis})}
                 secureTextEntry={this.state.pwdhide}
                 value={this.state.newpasswordbis}
               />
             </View>
          </View>

          <View>
            <TouchableOpacity onPress={this._pwdhide}>
              <Image
                style={{height: 40, width: 40, opacity: this.state.opacity}}
                source={require('./../../picture/start/eye.png')}
                resizeMode="contain"
                />
            </TouchableOpacity>
          </View>
          <Text style={{fontSize: 10}}>* le nouveau mot de passe doit contenir au minimum 8 caract√®res.</Text>


         </View>
         <View style={styles.containerFooter}>
           <Button
             title={'Retour'}
             style={styles.ButtonCo}
             color='#363453'
             onPress={this._cancelEdit}
           />
           <Button
             title={'Valider'}
             style={styles.ButtonCo}
             color='#363453'
             onPress={this._confirmEdit}
           />
         </View>

      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FFF',
    flex: 1
  },
  containerBody: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  input: {
    width: 220,
    height: 44,
    padding: 10,
    borderWidth: 1,
    borderColor: 'black',
    marginBottom: 10,
  },
  titleInfo: {
    color: '#363453',
    fontWeight: 'bold',
    fontSize: 14,
  },
  containerFooter: {
    flexDirection: 'row',
    paddingBottom: 10,
    justifyContent:'space-around',
  },
});
