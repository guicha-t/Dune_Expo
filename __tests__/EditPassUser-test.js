import EditPassUser from './../src/screen/userProfil/EditPassUser';
import 'isomorphic-fetch';
import React from 'react';

import renderer from 'react-test-renderer'

jest.useFakeTimers();

test('render EditPassUser correctly', () => {

    const tree = renderer.create(<EditPassUser />).toJSON();
    expect(tree).toMatchSnapshot();
});