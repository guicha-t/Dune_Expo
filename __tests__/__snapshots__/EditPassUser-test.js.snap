// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render EditPassUser correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "black",
          "elevation": 5,
          "flexDirection": "row",
          "height": 56,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
        },
        Object {
          "backgroundColor": "#f6c6d2",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 0.3,
          "justifyContent": "center",
          "paddingLeft": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 24,
              "width": 24,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0.4,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "flex": 1,
              "height": undefined,
              "width": undefined,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 0.3,
          "justifyContent": "center",
          "paddingRight": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 32,
              "width": 32,
            }
          }
        />
      </View>
    </View>
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderColor": "#ea4335",
              "borderRadius": 5,
              "borderWidth": 1,
              "marginHorizontal": 30,
              "maxWidth": 350,
              "shadowColor": "#000000",
              "shadowOpacity": 0.1,
              "shadowRadius": 10,
              "transform": Array [
                Object {
                  "scale": 0,
                },
              ],
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "justifyContent": "center",
                  "padding": 20,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#333",
                      "fontSize": 18,
                      "fontWeight": "500",
                      "lineHeight": 25,
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                ATTENTION !
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#666",
                      "fontSize": 17,
                      "paddingTop": 10,
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                Vous devez accepter les conditions d'utilisation avant de pouvoir utiliser l'application.
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderRadius": 5,
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#ea4335",
                    "borderRadius": 5,
                    "marginBottom": 20,
                    "marginHorizontal": 10,
                    "minWidth": 90,
                    "opacity": 1,
                    "paddingVertical": 8,
                  }
                }
                testID="buttonConfirm"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFF",
                        "fontSize": 15,
                        "fontWeight": "600",
                        "textAlign": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  COMPRIS
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "center",
          "marginBottom": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.3,
            "paddingBottom": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2A5E85",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Ancien
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 0.7,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Mot de passe"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "black",
              "borderWidth": 1,
              "height": 44,
              "marginBottom": 10,
              "padding": 10,
              "width": 220,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "center",
          "marginBottom": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.3,
            "paddingBottom": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2A5E85",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Nouveau*
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 0.7,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Mot de passe"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "black",
              "borderWidth": 1,
              "height": 44,
              "marginBottom": 10,
              "padding": 10,
              "width": 220,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "center",
          "marginBottom": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.3,
            "paddingBottom": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2A5E85",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Nouveau*
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 0.7,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Mot de passe"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "black",
              "borderWidth": 1,
              "height": 44,
              "marginBottom": 10,
              "padding": 10,
              "width": 220,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 40,
              "opacity": 1,
              "width": 40,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "fontSize": 10,
        }
      }
    >
      * le nouveau mot de passe doit contenir au minimum 8 caractères.
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "paddingBottom": 10,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#2A5E85",
              },
            ]
          }
        >
          Retour
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#2A5E85",
              },
            ]
          }
        >
          Valider
        </Text>
      </View>
    </View>
  </View>
</View>
`;
